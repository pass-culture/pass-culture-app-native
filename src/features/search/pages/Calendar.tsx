import React from 'react'
import { Text, View } from 'react-native'
import { CalendarList, LocaleConfig } from 'react-native-calendars'

import { Spacer } from 'ui/theme'

LocaleConfig.locales['fr'] = {
  monthNames: [
    'Janvier',
    'Février',
    'Mars',
    'Avril',
    'Mai',
    'Juin',
    'Juillet',
    'Août',
    'Septembre',
    'Octobre',
    'Novembre',
    'Décembre',
  ],
  monthNamesShort: [
    'Janv.',
    'Févr.',
    'Mars',
    'Avril',
    'Mai',
    'Juin',
    'Juil.',
    'Août',
    'Sept.',
    'Oct.',
    'Nov.',
    'Déc.',
  ],
  dayNames: ['Dimanche', 'Lundi', 'Mardi', 'Mercredi', 'Jeudi', 'Vendredi', 'Samedi'],
  dayNamesShort: ['Dim.', 'Lun.', 'Mar.', 'Mer.', 'Jeu.', 'Ven.', 'Sam.'],
}
LocaleConfig.defaultLocale = 'fr'
export const Calendar: React.FC = () => (
  <React.Fragment>
    <Spacer.TopScreen />
    <CalendarList
      minDate={'2021-02-02'}
      maxDate={'2021-12-30'}
      onDayPress={(day) => {
        console.log('selected day', day)
      }}
      monthFormat={'MMMM yyyy'}
      hideDayNames={true}
      firstDay={1}
      disableAllTouchEventsForDisabledDays={true}
      // renderHeader={(date) => <Text>Mois</Text>}
      pastScrollRange={0}
      futureScrollRange={50}
      scrollEnabled={true}
      showScrollIndicator={true}
      markedDates={{
        '2021-02-20': { disabled: true },
      }}
      renderHeader={(date) => <Text>{`${date.getMonth()} ${date.getFullYear()}`}</Text>}
      dayComponent={({ date, state, marking }) => {
        console.log(marking)
        return (
          // <View>
          //   <Text
          //     style={{
          //       textAlign: 'center',
          //       textDecorationLine: 'line-through',
          //       color: state === 'disabled' || marking['disabled'] === true ? 'gray' : 'black',
          //     }}>
          //     {date.day}
          //   </Text>
          //   <Text
          //     style={{
          //       textAlign: 'center',
          //       color: state === 'disabled' || marking['disabled'] === true ? 'gray' : 'black',
          //       fontSize: 10,
          //     }}>
          //     2€
          //   </Text>
          // </View>
          <React.Fragment>
            {state === 'disabled' || marking['disabled'] === true ? (
              <View
                style={{
                  position: 'absolute',
                  transform: [{ rotate: '135deg' }],
                  top: 8,
                  width: 28,
                  height: 1,
                  borderBottomColor: 'gray',
                  borderBottomWidth: 1,
                }}
              />
            ) : null}
            <Text>{date.day}</Text>
            <Text>19.90€</Text>
          </React.Fragment>
        )
      }}
    />
  </React.Fragment>
)
