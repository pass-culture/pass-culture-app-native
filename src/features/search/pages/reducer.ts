import { FetchAlgoliaParameters, LocationType, AlgoliaGeolocation } from 'libs/algolia'
import { DATE_FILTER_OPTIONS } from 'libs/algolia/enums'
import { SuggestedPlace } from 'libs/place'

import { clampPrice, addOrRemove } from './reducer.helpers'

export type SearchParameters = Omit<
  FetchAlgoliaParameters,
  'hitsPerPage' | 'page' | 'sortBy' | 'keywords'
>
export type SearchState = SearchParameters & { showResults: boolean }

export const initialSearchState: SearchState = {
  aroundRadius: null,
  offerCategories: [],
  tags: [],
  offerIsDuo: false,
  offerIsFree: false,
  offerIsNew: false,
  offerTypes: {
    isDigital: false,
    isEvent: false,
    isThing: false,
  },
  beginningDatetime: null,
  endingDatetime: null,
  priceRange: null,
  locationType: LocationType.EVERYWHERE,
  geolocation: null,
  date: null,
  timeRange: null,
  showResults: false,
  place: null,
}

export const DEFAULT_TIME_RANGE = [8, 24]

export type Action =
  | { type: 'INIT' }
  | { type: 'INIT_FROM_SEE_MORE'; payload: Partial<SearchState> }
  | { type: 'PRICE_RANGE'; payload: SearchState['priceRange'] }
  | { type: 'TIME_RANGE'; payload: SearchState['timeRange'] }
  | { type: 'OFFER_TYPE'; payload: keyof SearchState['offerTypes'] }
  | { type: 'SHOW_RESULTS'; payload: boolean }
  | { type: 'SET_CATEGORY'; payload: SearchState['offerCategories'] }
  | { type: 'TOGGLE_CATEGORY'; payload: string }
  | { type: 'TOGGLE_OFFER_FREE' }
  | { type: 'TOGGLE_OFFER_DUO' }
  | { type: 'TOGGLE_OFFER_NEW' }
  | { type: 'TOGGLE_DATE' }
  | { type: 'TOGGLE_HOUR' }
  | { type: 'SELECT_DATE_FILTER_OPTION'; payload: DATE_FILTER_OPTIONS }
  | { type: 'SELECT_DATE'; payload: Date }
  | { type: 'LOCATION_AROUND_ME'; payload: AlgoliaGeolocation }
  | { type: 'LOCATION_EVERYWHERE' }
  | { type: 'LOCATION_PLACE'; payload: SuggestedPlace }

export const searchReducer = (state: SearchState, action: Action): SearchState => {
  switch (action.type) {
    case 'INIT':
      return initialSearchState
    case 'SHOW_RESULTS':
      return { ...state, showResults: action.payload }
    case 'INIT_FROM_SEE_MORE':
      return {
        ...state,
        ...action.payload,
        priceRange: clampPrice(action.payload.priceRange),
      }
    case 'PRICE_RANGE':
      return { ...state, priceRange: action.payload }
    case 'TIME_RANGE':
      return { ...state, timeRange: action.payload }
    case 'TOGGLE_CATEGORY':
      return {
        ...state,
        offerCategories: addOrRemove(state.offerCategories, action.payload),
      }
    case 'SET_CATEGORY':
      return { ...state, offerCategories: action.payload }
    case 'OFFER_TYPE':
      return {
        ...state,
        offerTypes: { ...state.offerTypes, [action.payload]: !state.offerTypes[action.payload] },
      }
    case 'TOGGLE_OFFER_FREE':
      return { ...state, offerIsFree: !state.offerIsFree }
    case 'TOGGLE_OFFER_DUO':
      return { ...state, offerIsDuo: !state.offerIsDuo }
    case 'TOGGLE_OFFER_NEW':
      return { ...state, offerIsNew: !state.offerIsNew }
    case 'TOGGLE_DATE':
      if (state.date) return { ...state, date: null }
      return {
        ...state,
        date: {
          option: DATE_FILTER_OPTIONS.TODAY,
          selectedDate: new Date(),
        },
      }
    case 'TOGGLE_HOUR':
      if (state.timeRange) return { ...state, timeRange: null }
      return {
        ...state,
        timeRange: [8, 24],
      }
    case 'SELECT_DATE_FILTER_OPTION':
      if (!state.date) return state
      return { ...state, date: { ...state.date, option: action.payload } }
    case 'SELECT_DATE':
      if (!state.date) return state
      return { ...state, date: { ...state.date, selectedDate: action.payload } }
    case 'LOCATION_AROUND_ME':
      return {
        ...state,
        locationType: LocationType.AROUND_ME,
        geolocation: action.payload,
        place: null,
      }
    case 'LOCATION_EVERYWHERE':
      return {
        ...state,
        locationType: LocationType.EVERYWHERE,
        geolocation: null,
        place: null,
      }
    case 'LOCATION_PLACE':
      return {
        ...state,
        locationType: LocationType.PLACE,
        geolocation: action.payload.geolocation,
        place: action.payload,
      }
    default:
      return state
  }
}
