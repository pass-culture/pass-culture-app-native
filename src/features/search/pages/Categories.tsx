import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import debounce from 'lodash.debounce'
import React, { useRef } from 'react'
import styled from 'styled-components/native'

import { useSearch } from 'features/search/pages/SearchWrapper'
import { CATEGORY_CRITERIA } from 'libs/algolia/enums'
import { _ } from 'libs/i18n'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { Validate } from 'ui/svg/icons/Validate'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY } from 'ui/theme/colors'

const ALL = 'ALL'
export const DEBOUNCED_CALLBACK = 500

export const useSelectCategory = (callback: () => void) => {
  const { searchState, dispatch } = useSearch()
  const debouncedCallback = useRef(debounce(callback, DEBOUNCED_CALLBACK)).current

  return {
    isCategorySelected: (category: string) => {
      const [selectedCategory] = [...searchState.offerCategories, ALL]
      return selectedCategory === category
    },
    selectCategory: (category: string) => () => {
      const payload = category === ALL ? [] : [category]
      dispatch({ type: 'SET_CATEGORY', payload })
      debouncedCallback()
    },
  }
}

export const Categories: React.FC = () => {
  const { goBack } = useNavigation()
  const { isCategorySelected, selectCategory } = useSelectCategory(goBack)

  return (
    <React.Fragment>
      <React.Fragment>
        <Container>
          <Spacer.TopScreen />
          <Spacer.Column numberOfSpaces={16} />

          {Object.entries(CATEGORY_CRITERIA).map(([category, { label, icon: Icon }]) => {
            const isSelected = isCategorySelected(category)
            const color2 = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.SECONDARY
            const textColor = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.BLACK

            return (
              <LabelContainer key={category} onPress={selectCategory(category)} testID={category}>
                <Spacer.Row numberOfSpaces={4} />
                <Icon size={getSpacing(12)} color={ColorsEnum.PRIMARY} color2={color2} />
                <Spacer.Row numberOfSpaces={2} />
                <Typo.ButtonText color={textColor}>{label}</Typo.ButtonText>
                <Spacer.Flex />
                {isSelected && <Validate color={ColorsEnum.PRIMARY} size={getSpacing(8)} />}
                <Spacer.Row numberOfSpaces={6} />
              </LabelContainer>
            )
          })}

          <Spacer.Column numberOfSpaces={30} />
        </Container>
      </React.Fragment>

      <PageHeader title={_(t`CatÃ©gories`)} />
    </React.Fragment>
  )
}

const Container = styled.ScrollView({ flex: 1 })

const LabelContainer = styled.TouchableOpacity.attrs({
  activeOpacity: ACTIVE_OPACITY,
})({
  flexDirection: 'row',
  alignItems: 'center',
  marginBottom: getSpacing(4),
})
