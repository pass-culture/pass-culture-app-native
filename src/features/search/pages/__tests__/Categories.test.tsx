import { render } from '@testing-library/react-native'
import React from 'react'

import { initialSearchState } from 'features/search/pages/reducer'

import { Categories } from '../Categories'

const mockSearchState = initialSearchState

jest.mock('features/search/pages/SearchWrapper', () => ({
  useSearch: () => ({
    searchState: mockSearchState,
    dispatch: jest.fn(),
  }),
}))

describe('Categories component', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<Categories />)
    expect(toJSON()).toMatchSnapshot()
  })
  it('should match diff snapshot when new category is selected', () => {
    const allSelected = render(<Categories />).toJSON()

    mockSearchState.offerCategories = ['CINEMA']
    const cinemaSelected = render(<Categories />).toJSON()
    expect(cinemaSelected).toMatchDiffSnapshot(allSelected)
  })
})
