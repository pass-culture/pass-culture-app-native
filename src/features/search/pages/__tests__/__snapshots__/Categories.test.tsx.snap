// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component should match diff snapshot when new category is selected 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -76,15 +76,15 @@
                },
              ]
            }
          />
          <Text
-           color=\\"#151515\\"
+           color=\\"#eb0055\\"
            style={
              Array [
                Object {
-                 \\"color\\": \\"#151515\\",
+                 \\"color\\": \\"#eb0055\\",
                  \\"fontFamily\\": \\"Montserrat-Bold\\",
                  \\"fontSize\\": 15,
                  \\"lineHeight\\": 20,
                },
              ]
@@ -101,10 +101,16 @@
                  \\"flexShrink\\": 1,
                },
              ]
            }
          />
+         <View
+           height={32}
+           width={32}
+         >
+           undefined-SVG-Mock
+         </View>
          <View
            numberOfSpaces={6}
            style={
              Array [
                Object {
@@ -159,15 +165,15 @@
                },
              ]
            }
          />
          <Text
-           color=\\"#eb0055\\"
+           color=\\"#151515\\"
            style={
              Array [
                Object {
-                 \\"color\\": \\"#eb0055\\",
+                 \\"color\\": \\"#151515\\",
                  \\"fontFamily\\": \\"Montserrat-Bold\\",
                  \\"fontSize\\": 15,
                  \\"lineHeight\\": 20,
                },
              ]
@@ -184,16 +190,10 @@
                  \\"flexShrink\\": 1,
                },
              ]
            }
          />
-         <View
-           height={32}
-           width={32}
-         >
-           undefined-SVG-Mock
-         </View>
          <View
            numberOfSpaces={6}
            style={
              Array [
                Object {"
`;

exports[`Categories component should render correctly 1`] = `
Array [
  <RCTScrollView
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View>
      <View
        customHeight={16}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <View
        numberOfSpaces={16}
        style={
          Array [
            Object {
              "height": 64,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="ALL"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#eb0055"
          style={
            Array [
              Object {
                "color": "#eb0055",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Toutes les catégories
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          height={32}
          width={32}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="CINEMA"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Cinéma
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="VISITE"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Visites, expositions
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="MUSIQUE"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Musique
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="SPECTACLE"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Spectacles
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="LECON"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Cours, ateliers
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="LIVRE"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Livres
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="FILM"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Films, séries, podcasts
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="PRESSE"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Presse
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="JEUX_VIDEO"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Jeux vidéos
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="CONFERENCE"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Conférences, rencontres
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
        testID="INSTRUMENT"
      >
        <View
          numberOfSpaces={4}
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
        />
        <View
          height={48}
          width={48}
        >
          undefined-SVG-Mock
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <Text
          color="#151515"
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Instruments de musique
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "width": 24,
              },
            ]
          }
        />
      </View>
      <View
        numberOfSpaces={30}
        style={
          Array [
            Object {
              "height": 120,
            },
          ]
        }
      />
    </View>
  </RCTScrollView>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eb0055",
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      customHeight={16}
      style={
        Array [
          Object {
            "height": 16,
          },
        ]
      }
    />
    <View
      numberOfSpaces={2}
      style={
        Array [
          Object {
            "height": 8,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        numberOfSpaces={5}
        style={
          Array [
            Object {
              "width": 20,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          height={32}
          testID="icon-back"
          width={32}
        >
          icon-back-SVG-Mock
        </View>
      </View>
      <View
        numberOfSpaces={0}
        style={
          Array [
            Object {
              "width": 0,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "flexShrink": 1,
              "textAlign": "center",
            },
          ]
        }
      >
        <Text
          color="#ffffff"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 15,
                "lineHeight": 20,
              },
            ]
          }
        >
          Catégories
        </Text>
      </Text>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
      <View
        numberOfSpaces={10}
        style={
          Array [
            Object {
              "width": 40,
            },
          ]
        }
      />
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "width": 24,
            },
          ]
        }
      />
    </View>
    <View
      numberOfSpaces={2}
      style={
        Array [
          Object {
            "height": 8,
          },
        ]
      }
    />
  </View>,
]
`;
